contracts/bonding/BondingManager.sol

```diff
@@ -546,6 +559,43 @@ contract BondingManager is ManagerProxyTarget, IBondingManager {
         }
 
         emit Bond(_to, currentDelegate, _owner, _amount, del.bondedAmount);
+
+        // the `autoCheckpoint` modifier has been replaced with its internal function as a `Stack too deep` error work-around
+        checkpointBondingState(_owner, del, transcoders[_to]);
```

Comment:
Wow yeah. I'll make sure to add some test checks to ensure this. The only reason it didn't affect them is because we only use the `lastRewardRound` from the `Transcoder` struct right now, which doesn't affect the bonding state calculation for delegators (cause we check the `delegateAddress` before that.

Reply:
Addressed on https://github.com/livepeer/protocol/pull/614/commits/fecf5a2e0e0cf76472ef1c23d543bc6f9512fb9f together with the test updates